src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/ArgumentObject.cs

```diff
@@ -0,0 +1,16 @@
+using System;
+
+namespace Microsoft.CSharp.RuntimeBinder
+{
+    // This class is used to keep the tuple of runtime object values and 
+    // the type that we want to use for the argument. This is different than the runtime
+    // value's type because unless the static time type was dynamic, we want to use the
+    // static time type. Also, we may have null values, in which case we would not be 
+    // able to get the type.
+    internal sealed class ArgumentObject
```

Comment:
I made a start on that, and maybe I'd just made a silly slip, but there were enough errors in running the tests that I think it should be left separate if it is worth doing. At least, it didn't fit with the approach I'm taking here of lots of very small changes were each is individually very low impact.